core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -220,4 +223,30 @@ boolean isMemoryLocked() {
 
     }
 
+    static class MaxNumberOfThreadsCheck implements Check {
+
+        private static final long MAX_FILES_PER_THREADS_THRESHOLD = 1 << 15;
```

Comment:
Technically that would be a "brain fart"; I pushed 34d7dfcf2dcb99a0853f73d5cfe688af138cdcfc.